pkg/msr/msr_linux.go

```diff
@@ -8,20 +8,66 @@ package msr
 import (
 	"encoding/binary"
 	"fmt"
+	"io/ioutil"
 	"os"
 	"path/filepath"
 	"strconv"
+	"strings"
 )
 
 // CPUs is a slice of the various cpus to read or write the MSR to.
 type CPUs []uint64
 
+func parseCPUs(s string) (CPUs, error) {
+	cpus := make(CPUs, 0)
+	// We expect the format to be "0-5,7-8..." or we could also get just one cpu.
+	// We're unlikely to get more than one range since we're looking at present cpus,
+	// but handle it just in case.
+	ranges := strings.Split(strings.TrimSpace(s), ",")
+	for _, r := range ranges {
+		if len(r) == 0 {
+			continue
+		}
+		// Split on a - if it exists.
+		cs := strings.Split(r, "-")
```

Comment:
I think you're good on stupid stuff like - by itself, by what about -1? should that be an error?

Reply:
I see. I'll check for this

Reply:
That gets caught by the parseuint